### ISSUE TITLE
Cropbox widget not working in modal window 

### ISSUE THREAD

**Issue by Geser1979** [bupy7/yii2-widget-cropbox#15]
Please help. In modal window cannot upload and crop image file.

**Comment by bupy7** [comment]
Give me your code, please, and version yii2-widget-cropbox.

**Comment by Geser1979** [comment]
Version "bupy7/yii2-widget-cropbox": "2.2"

JS for activate modal
function updateprofile(type, attorneyid){
    $.ajax({
          type:'POST',
          url:'index.php?r=controller%2Faction&Id='+id+'&type='+type,
          success: function(data)
                   {
                       $('#myModal').html(data);
                       $('#myModal').modal();
                   }
      });
   }

In main layout
<div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true"></div>

In controller
$model = Model::find()->where('(id = :id)', [':id' => $Id,])->one();
        if (Yii::$app->request->isAjax && $attorney->load(Yii::$app->request->post())) {
            if ($model->validate()) {
                $model->image = \yii\web\UploadedFile::getInstance($model, 'image');
            $model->save();
            return $this->redirect(['/account/profile']);
            } 
            Yii::$app->response->format = 'json';
                return \yii\widgets\ActiveForm::validate($model);
        }
        
        return $this->renderAjax('popUpUpdate', [
            'model' => $model,
        ]);

In view
<?php
/* @var $this yii\web\View */
/* @var $form yii\bootstrap\ActiveForm */
/* @var $model app\models\Response */
use yii\helpers\Html;
use yii\bootstrap\ActiveForm;
use bupy7\cropbox\Cropbox;
use kartik\checkbox\CheckboxX;
?>

<div class="modal-dialog">
<div class="modal-content">
<div class="modal-header">
    <button type="button" class="close" data-dismiss="modal" ><span aria-hidden="true" style="color:red;">X</span><span class="sr-only">Close</span></button>
</div>
    
<div class="modal-body">
    <div class="form-group">
    <h4 class="modal-title" id="myModalLabel">Edit <?php echo $typeLable[$type]; ?></h4>
    </div>
    
    <?php $form = ActiveForm::begin(['id' => 'updateprofile-form',     'enableClientValidation' => true,
        'enableAjaxValidation' => true,
    'validateOnChange' => false,
    'validateOnBlur' => false,
    'validateOnSubmit' => true,
        'layout' => 'horizontal',
            'fieldConfig' => [
            'template' => "{label}\n{beginWrapper}\n{input}\n{hint}\n{error}\n{endWrapper}",
            'horizontalCssClasses' => [
                'label' => 'col-sm-3',
                'offset' => 'col-sm-offset-3',
                //'wrapper' => 'input-required col-sm-3',
                'error' => '',
                'hint' => 'col-sm-3',
            ],
        ],]); ?>
        <?= $form->field($model, $type, ['horizontalCssClasses' => [
                                  'wrapper' => 'col-sm-7',
                                ]])->widget(Cropbox::className(), [
            'attributeCropInfo' => 'crop_info',
        ]); ?>
    <div class="modal-footer">
   <div class="form-group our-form-group">
        <?= Html::submitButton('Save', ['class' => 'btn btn-primary', 'name' => 'placebid-button']) ?>
   </div>
   <?php ActiveForm::end(); ?>
</div>
</div>
</div>

**Comment by bupy7** [comment]
What specifically is not working?

**Comment by Geser1979** [comment]
In modal window cannot upload and crop image. See code above.

**Comment by bupy7** [comment]
@Geser1979 Attach browser console log after it actions.

**Comment by Geser1979** [comment]
After action in console i see this
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.

**Comment by bupy7** [comment]
You can attach screenshot of console log?

**Comment by bupy7** [comment]
You load `yii2-widget-cropbox` via Ajax. It is not related to his work. You sure that all JS scripts of `yii2-widget-cropbox` has been loaded?

**Comment by Geser1979** [comment]
Yes, all JS scripts has been loaded. So widget cannot work via Ajax?

**Comment by bupy7** [comment]
Can do it. But I not understand that is problem. Try set custom `id` of widget as below:
```php
echo $form->field($model, 'image')->widget(Cropbox::className(), [
    'attributeCropInfo' => 'crop_info',
    'id' => uniqid(),
]);
```

**Comment by bupy7** [comment]
@Geser1979 Yes, if you uses other widgets before call the modal with `yii2-widget-cropbox`, then most likely ID of `#w{N}` already busy and `yii2-widget-cropbox` adds event for `#w{N}` element. You need easy make unique ID of `yii2-widget-cropbox`.

**Comment by Geser1979** [comment]
Yes, now widget works in modal window, but after submit there is nothing in GLOBAL _FILES and image does not update. Are there another settings for this?

**Comment by bupy7** [comment]
```php
use bupy7\cropbox\Cropbox;

$form = ActiveForm::begin([
    'options' => ['enctype'=>'multipart/form-data'],
]);

...

echo $form->field($model, 'image')->widget(Cropbox::className(), [
    ...
]);

...
```

**Comment by Geser1979** [comment]
All works! Thanks!
